nats.go

```diff
@@ -2266,6 +2283,14 @@ func (s *Subscription) SetPendingLimits(msgLimit, bytesLimit int) error {
 	return nil
 }
 
+// SetSlowConsumerTimeout sets the deadline to wait before dropping mesages
+// when mode of delivery for subscription is a channel.
+func (s *Subscription) SetSlowConsumerTimeout(timeout time.Duration) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.sct = timeout
```

Comment:
Should there be checks for validity and to see if this is a channel subscription, as in `SetPendingLimits`?

```
 	if s.conn == nil {
 		return ErrBadSubscription
 	}
 	if s.typ != ChanSubscription {
 		return ErrTypeSubscription
 	}
```